block/manager.go

```diff
@@ -497,6 +497,9 @@ func (m *Manager) retrieveBatch(ctx context.Context) (*BatchData, error) {
 			m.logger.Error("error while setting last batch hash", "error", err)
 		}
 		m.lastBatchData = res.BatchData
```

Comment:
i would move this after the check. `lastBatchData` is only valid for batches on which you were able to publish the headers. unless you have the case where headers are published for empty batches?